tests/nexus_workflow_test.go

```diff
@@ -115,3 +127,155 @@ func (s *ClientFunctionalSuite) TestNexusScheduleAndCancelCommands() {
+		TaskQueue: taskQueue,
+	}, "workflow")
+	s.NoError(err)
+	pollResp, err := s.engine.PollWorkflowTaskQueue(ctx, &workflowservice.PollWorkflowTaskQueueRequest{
+		Namespace: namespace,
+		TaskQueue: &taskqueue.TaskQueue{
+			Name: taskQueue,
+			Kind: enums.TASK_QUEUE_KIND_NORMAL,
+		},
+		Identity: "test",
+	})
+	s.NoError(err)
+	_, err = s.engine.RespondWorkflowTaskCompleted(ctx, &workflowservice.RespondWorkflowTaskCompletedRequest{
+		Identity:  "test",
+		TaskToken: pollResp.TaskToken,
+		Commands: []*commandpb.Command{
+			{
+				CommandType: enums.COMMAND_TYPE_SCHEDULE_NEXUS_OPERATION,
+				Attributes: &commandpb.Command_ScheduleNexusOperationCommandAttributes{
+					ScheduleNexusOperationCommandAttributes: &commandpb.ScheduleNexusOperationCommandAttributes{
+						Service:   serviceName,
+						Operation: "operation",
+						Input:     s.mustToPayload("input"),
+					},
+				},
+			},
+		},
+	})
+	s.NoError(err)
+	pollResp, err = s.engine.PollWorkflowTaskQueue(ctx, &workflowservice.PollWorkflowTaskQueueRequest{
+		Namespace: namespace,
+		TaskQueue: &taskqueue.TaskQueue{
+			Name: taskQueue,
+			Kind: enums.TASK_QUEUE_KIND_NORMAL,
+		},
+		Identity: "test",
+	})
+	s.NoError(err)
+	completedEventIdx := slices.IndexFunc(pollResp.History.Events, func(e *historypb.HistoryEvent) bool {
+		return e.GetNexusOperationCompletedEventAttributes() != nil
+	})
+	s.Greater(completedEventIdx, 0)
+
+	_, err = s.engine.RespondWorkflowTaskCompleted(ctx, &workflowservice.RespondWorkflowTaskCompletedRequest{
+		Identity:                   "test",
+		ReturnNewWorkflowTask:      true,
+		ForceCreateNewWorkflowTask: true,
+		TaskToken:                  pollResp.TaskToken,
+		Commands: []*commandpb.Command{
+			{
+				CommandType: enums.COMMAND_TYPE_COMPLETE_WORKFLOW_EXECUTION,
+				Attributes: &commandpb.Command_CompleteWorkflowExecutionCommandAttributes{
+					CompleteWorkflowExecutionCommandAttributes: &commandpb.CompleteWorkflowExecutionCommandAttributes{
+						Result: &commonpb.Payloads{
+							Payloads: []*commonpb.Payload{
+								pollResp.History.Events[completedEventIdx].GetNexusOperationCompletedEventAttributes().Result,
+							},
+						},
+					},
+				},
+			},
+		},
+	})
+	s.NoError(err)
+	var result string
+	s.NoError(run.Get(ctx, &result))
+	s.Equal("result", result)
+}
+
+func allocListenAddress(t *testing.T) string {
+	pp := temporalite.NewPortProvider()
+	listenAddr := fmt.Sprintf("localhost:%d", pp.MustGetFreePort())
+	require.NoError(t, pp.Close())
+	return listenAddr
+}
+
+func newNexusServer(t *testing.T, listenAddr string, handler nexus.Handler) func() error {
+	hh := nexus.NewHTTPHandler(nexus.HandlerOptions{
+		Handler: handler,
+	})
+	srv := &http.Server{Addr: listenAddr, Handler: hh}
+	listener, err := net.Listen("tcp", listenAddr)
+	require.NoError(t, err)
+	errCh := make(chan error, 1)
+	go func() {
+		errCh <- srv.Serve(listener)
+	}()
+
+	return func() error {
```

Comment:
nit: you could register this with `t.Cleanup` so your tests don't need to care about calling it